droidlet/memory/memory_nodes.py

```diff
@@ -784,6 +785,36 @@ class SelfNode(PlayerNode):
     TABLE_COLUMNS = ["uuid", "eid", "name", "x", "y", "z", "pitch", "yaw", "ref_type"]
     NODE_TYPE = "Self"
 
+    @classmethod
+    def create(cls, memory, player_struct=None, memid=None) -> str:
+        """Creates a new entry into the ReferenceObjects table
+
+        Returns:
+            string: memid of the entry
+
+        """
+        memid = memid or cls.new(memory)
+        if player_struct is None:
+            eid, name, x, y, z, pitch, yaw = None, None, None, None, None, None, None
+        else:
+            eid, name = player_struct.entityId, player_struct.name
+            x, y, z = player_struct.pos
+            yaw, pitch = player_struct.look
+        cmd = "INSERT INTO ReferenceObjects(uuid, eid, name, x, y, z, pitch, yaw, ref_type) VALUES (?,?,?,?,?,?,?,?,?)"
```

Comment:
to be clear, earlier we were doing this for the agents here: https://github.com/facebookresearch/fairo/blob/main/droidlet/memory/sql_memory.py#L140 
which was equivalent to doing this anyway, right ?

Reply:
it didn't set it as a ReferenceObject; that was being done in the agent.